includes/admin/class-wc-stripe-admin-notices.php

```diff
@@ -174,8 +174,10 @@ public function stripe_check_environment() {
 
 			if ( empty( $show_keys_notice ) ) {
 				$secret = WC_Stripe_API::get_secret_key();
+				// if '_wcstripe_feature_upe_settings' is enabled show the notice in main settings page only
+				$should_show_notice_on_page = WC_Stripe_Feature_Flags::is_upe_settings_redesign_enabled() ? ( isset( $_GET['page'] ) && ! isset( $_GET['section'] ) ) : ! ( isset( $_GET['page'], $_GET['section'] ) && 'wc-settings' === $_GET['page'] && 'stripe' === $_GET['section'] );
```

Comment:
Good catch. I will fix this :+1:

Reply:
Fixed the bug in [this commit](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1774/commits/c514af44097b6a716736e872dc3907ec8c8e09c5). 

Instead of matching the `stripe_` substring, matched the `stripe` substring so that all the other gateways as well as the stripe itself is covered in the logic.

Reply:
Good call! It doesn't make sense to display it for the main gateway too.

Sorry to be so picky and this is technically correct but it is a pretty long condition that requires some grokking. And my bad for mentioning `substr()` - I actually had `strpos()` in mind!

How about this instead? Let me know if I missed something that your idea covers that this one doesn't!

```php
$is_stripe_payment_method_page = isset( $_GET['page'], $_GET['section'] ) && 'wc-settings' === $_GET['page'] && 0 === strpos( $_GET['section'], 'stripe' );
$should_show_notice_on_page    = WC_Stripe_Feature_Flags::is_upe_settings_redesign_enabled() && ! $is_stripe_payment_method_page;
```

Reply:
I'm looking at this again condition again and I get the idea now - if the feature flag then hide the notice from the main Stripe gateway and the payment method pages, and if the flag is disabled - just the Stripe gateway? Makes sense! :)

I don't think the other filters are necessary though - there's no harm if someone tries to pass a malicious value since the code is only checking if it equals something (and not, e.g., displaying it or feeding it into a DB) so I'd just go with:

```php
$is_stripe_payment_method_page = isset( $_GET['page'], $_GET['section'] ) && 'wc-settings' === $_GET['page'] && 0 === strpos( $_GET['section'], 'stripe' );
$is_stripe_main_gateway_page   = isset( $_GET['page'], $_GET['section'] ) && 'wc-settings' === $_GET['page'] && 'stripe' === $_GET['section'];
$should_show_notice_on_page    = WC_Stripe_Feature_Flags::is_upe_settings_redesign_enabled() ? ! $is_stripe_payment_method_page : $is_stripe_main_gateway_page;
```

You can probably even simplify it to not distinguish the cases when UPE is enabled and just doing:

```php
$should_show_notice_on_page = isset( $_GET['page'], $_GET['section'] ) && 'wc-settings' === $_GET['page'] && 0 === strpos( $_GET['section'], 'stripe' );
```